app/components/primer/button_component.rb

```diff
@@ -56,6 +56,21 @@ class ButtonComponent < Primer::Component
     }
     alias counter trailing_visual_counter # remove alias when all buttons are migrated to new slot names
 
+    # `Tooltip` that appears on mouse hover or keyboard focus over the button. Use tooltips sparingly and as a last resort.
+    # Consult the <%= link_to_component(Primer::Alpha::Tooltip) %> documentation for more information.
+    #
+    # @param system_arguments [Hash] Same arguments as <%= link_to_component(Primer::Alpha::Tooltip) %>.
+    renders_one :tooltip, lambda { |**system_arguments|
```

Comment:
```suggestion
    renders_one :tooltip, lambda { |text:, **system_arguments|
```

We may want to explicitly surface the required `text:` argument here. If we do this, we'll want to add `@param` to document this.

Reply:
I thought about this, but then I saw `type` and `for_id` are required too.

I decided keeping the definition of which parameters are required on `Tooltip` will be easier to maintain. I'll leave it as is for now if that's ok.

Reply:
That makes sense 👍